dungeon/views.py

```diff
@@ -7,25 +7,26 @@
 from dungeon.models import Npc
 from dungeon.serializers import NpcSerializer
 from rest_framework.decorators import api_view
+from rest_framework.response import Response
 
 @api_view(['GET', 'POST', 'DELETE'])
 def npc_list(request):
     if request.method == 'GET':
         npcs = Npc.objects.all()
 
         npc_serializer = NpcSerializer(npcs, many=True)
-        return JsonResponse(npc_serializer.data, safe=False)
+        return Response(npc_serializer.data)
 
     elif request.method == 'POST':
         npc_data = JSONParser().parse(request)
         npc_serializer = NpcSerializer(data=npc_data)
         if npc_serializer.is_valid():
             npc_serializer.save()
-            return JsonResponse(npc_serializer.data, status=status.HTTP_201_CREATED)
-        return JsonResponse(npc_serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+            return Response(npc_serializer.data, status=status.HTTP_201_CREATED)
+        return Response(npc_serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
What is the difference between the two?

Reply:
Jsonresponse when look at by testing only returns the status code and a reference to the actual json. Response returns that actually data

Reply:
ahh thank you for explaining